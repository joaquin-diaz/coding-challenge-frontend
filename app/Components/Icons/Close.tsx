import styled from '@emotion/styled';
import styles from '../../constants/styles';

const Close = styled.div`
  width: 32px;
  height: 32px;

  &:before,
  &:after {
    position: absolute;
    left: 15px;
    content: ' ';
    height: 33px;
    width: 2px;
    background-color: ${styles.colors.primary};
  }

  &:before {
    transform: rotate(45deg);
  }

  &:after {
    transform: rotate(-45deg);
  }
`;

export default Close;
