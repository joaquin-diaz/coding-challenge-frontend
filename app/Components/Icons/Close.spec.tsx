import * as React from 'react';

import Close from './Close';
import { mount } from 'enzyme';

describe('<Close />', () => {
  it('should render', () => {
    const wrapper = mount(<Close />);
    expect(wrapper).toMatchSnapshot();
  });
});
