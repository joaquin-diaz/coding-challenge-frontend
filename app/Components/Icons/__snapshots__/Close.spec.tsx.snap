// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Close /> should render 1`] = `
.emotion-0 {
  width: 32px;
  height: 32px;
}

.emotion-0:before,
.emotion-0:after {
  position: absolute;
  left: 15px;
  content: ' ';
  height: 33px;
  width: 2px;
  background-color: #3f51b5;
}

.emotion-0:before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.emotion-0:after {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<Styled(div)>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;
